dash/include/dash/Graph.h

```diff
@@ -0,0 +1,876 @@
+#ifndef DASH__GRAPH_H__INCLUDED
+#define DASH__GRAPH_H__INCLUDED
+
+#include <vector>
+#include <unordered_set>
+#include <unordered_map>
+#include <map>
+#include <dash/graph/VertexIterator.h>
+#include <dash/graph/EdgeIterator.h>
+#include <dash/graph/InEdgeIterator.h>
+#include <dash/graph/OutEdgeIterator.h>
+#include <dash/graph/internal/Graph.h>
+#include <dash/memory/GlobHeapContiguousMem.h>
+#include <dash/memory/GlobHeapCombinedMem.h>
+#include <dash/Team.h>
+#include <dash/internal/Math.h>
+
+namespace dash {
+
+/**
+ * \defgroup DashGraphConcept Graph Concept
+ * Concept of a distributed graph container.
+ * 
+ * \ingroup DashContainerConcept
+ * \{
+ * \psr Description
+ * 
+ * Description here.
+ * 
+ * Extends Concepts:
+ * 
+ * Container properties:
+ * 
+ * Iterator validity:
+ * 
+ * \par Member types
+ * 
+ * \par Member functions
+ * 
+ * \}
+ * 
+ * Usage examples:
+ * 
+ */
+
+/**
+ * Distributed, dynamic graph container for sparse graphs.
+ */
+template<
+  GraphDirection Direction  = DirectedGraph,
+  typename VertexProperties = EmptyProperties,  // user-defined struct
+  typename EdgeProperties   = EmptyProperties,  // user-defined struct
+  typename VertexSizeType   = int,
+  typename EdgeSizeType     = int,
+//template<class, class...> typename EdgeContainer   = std::vector,
+//template<class, class...> typename VertexContainer = std::vector
+  template<class, class...> class _EdgeContainer   = std::vector,
```

Comment:
order is reversed, above its always vertex, than edge, here it is edge, than vertex